nodejs/lancedb/embedding/registry.ts

```diff
@@ -0,0 +1,157 @@
+import { EmbeddingFunction } from "./embedding_function";
+import "reflect-metadata";
+export interface EmbeddingFunctionOptions {
+  [key: string]: unknown;
+}
+
+export interface EmbeddingFunctionFactory<
+  T extends EmbeddingFunction = EmbeddingFunction,
+> {
+  new (modelOptions?: EmbeddingFunctionOptions): T;
+}
+
+interface EmbeddingFunctionCreate<T extends EmbeddingFunction> {
+  create(options?: EmbeddingFunctionOptions): T;
+}
+
+/**
+ * This is a singleton class used to register embedding functions
+ * and fetch them by name. It also handles serializing and deserializing.
+ * You can implement your own embedding function by subclassing EmbeddingFunction
+ * or TextEmbeddingFunction and registering it with the registry
+ */
+export class EmbeddingFunctionRegistry {
+  #functions: Map<string, EmbeddingFunctionFactory> = new Map();
```

Comment:
What does the `#` mean in the name?

Reply:
the `#` means it's _actually_ private. unlike the typescript modifier of `private`, the `#` symbol makes it inaccessible from other classes

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties